src/mon/OSDMonitor.cc

```diff
@@ -3379,8 +3379,14 @@ bool OSDMonitor::handle_osd_timeouts(const utime_t &now,
       continue;
     const std::map<int,utime_t>::const_iterator t = last_osd_report.find(i);
     if (t == last_osd_report.end()) {
-      // it wasn't in the map; start the timer.
-      last_osd_report[i] = now;
+      // it must be down because leader hasn't received osd beacon for more than 'timeo'.
```

Comment:
is it possible that the osd is just added to the monmap, and has not sent any beacon to monitor?

Reply:
@tchaikov  Thank you for reminding me that case, is it more reasonable to use 'last_osd_report[i] = now' when the osd is just added to the osdmap? Make sure that the osd is in last_osd_report[] only if it exists in osdmap.

Reply:
That would also work.  The code is structured this way to be defensive: it's more difficult to audit all the paths by which  the osd might be added.  If we fixed them all, this could become an assertion.  I'm inclined to leave it as is, though, even if you fix the other paths.